### PR TITLE
Add: CI parameter to SSH connection for automatic skip of user input

### PR DESCRIPTION
**Pull request by y0urself**
## What

[Add: CI parameter to SSH connection for automatic skip of user input](https://github.com/greenbone/python-gvm/commit/d51a0d818d28b7764d9cdb585e5bdaedce001e37)

## Why

* It should be more easy to skip the user input for things like e.g. CI

### CODE REVIEW

File: gvm/connections.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ class GvmConnection(XmlReader):
 
     def __init__(self, timeout: Optional[int] = DEFAULT_TIMEOUT):
         self._socket = None
-        self._timeout = timeout if timeout is not None else DEFAULT_TIMEOUT
```
Comment (bjoernricks) [inline gvm/connections.py:100]: I am not sure if this is correct. A timeout of 0 has a special meaning and it wouldn't be possible to set it anymore.

File: gvm/connections.py [hunk 2]
```diff
@@ -194,6 +191,7 @@ def __init__(
         username: Optional[str] = DEFAULT_SSH_USERNAME,
         password: Optional[str] = DEFAULT_SSH_PASSWORD,
         known_hosts_file: Optional[str] = None,
+        ci: Optional[bool] = None,
```
[the remark this answers is not in the document]
Comment (y0urself) [reply gvm/connections.py:194]: Did not considered this ...

File: gvm/connections.py (hunk 1 above)
Comment (y0urself) [reply gvm/connections.py:100]: Did not considered this ...

File: tests/connections/test_ssh_connection.py [hunk 3]
```diff
@@ -77,6 +77,13 @@ def test_connect_error(self):
         with self.assertRaises(GvmError, msg="SSH Connection failed"):
             ssh_connection.connect()
 
+    def test_connect_error_ci(self):
```
Comment (bjoernricks) [inline tests/connections/test_ssh_connection.py:80]: Could you rename the tests too?

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**